src/Membership.sol

```diff
@@ -11,432 +11,353 @@ error InvalidRateLimit();
 
 // It's not possible to acquire the rate limit due to exceeding the expected limits
 // even after attempting to erase expired memberships
-error ExceedAvailableMaxRateLimitPerEpoch();
+error ExceededMaxTotalRateLimit();
 
-// This membership is not in grace period yet
+// This membership is not in grace period yet // FIXME: yet or also already?
 error NotInGracePeriod(uint256 idCommitment);
 
 // The sender is not the holder of the membership
-error NotHolder(uint256 idCommitment);
+error AttemptedExtensionByNonHolder(uint256 idCommitment);
 
-// This membership cannot be erased (either it is not expired or not in grace period and/or not the owner)
+// This membership cannot be erased (either it is not expired or not in grace period and/or not the owner) // FIXME:
```

Comment:
I think best to refer to https://github.com/waku-org/specs/blob/feat-rln-on-mainnet-spec-edits/standards/core/rln-contract.md#membership-lifecycle and indeed split.

- is active: `CannotEraseActiveMembership`
- is in grace period: `NonHolderCannotErase`
- is expired: anyone can erase